File: frontend/src/types/task.ts
export type TaskType = "story" | "bug" | "task" | "epic";
export type TaskStatus = "backlog" | "todo" | "in_progress" | "in_review" | "done";
export type TaskPriority = "low" | "medium" | "high" | "critical";
export type JiraSyncStatus = "pending" | "synced" | "failed" | "local_only";

export interface Task {
  id: string;
  project_id: string;
  title: string;
  description: string;
  type: TaskType;
  status: TaskStatus;
  priority: TaskPriority;
  estimated_points: number | null;
  actual_points: number | null;
  sprint_id: string | null;
  jira_key: string | null;
  jira_sync_status: JiraSyncStatus;
  tags: string[];
  created_at: string;
  updated_at: string;
}

export interface CreateTaskRequest {
  project_id: string;
  title: string;
  description: string;
  type?: TaskType;
  priority?: TaskPriority;
  estimated_points?: number;
  sprint_id?: string;
  tags?: string[];
}

export interface PatchTaskRequest {
  title?: string;
  description?: string;
  status?: TaskStatus;
  priority?: TaskPriority;
  estimated_points?: number;
  actual_points?: number;
  sprint_id?: string;
  tags?: string[];
}

File: backend/src/domain/entities/task.py
from __future__ import annotations

import uuid
from dataclasses import dataclass, field
from datetime import datetime
from enum import Enum
from typing import Optional


class TaskType(str, Enum):
    STORY = "story"
    BUG = "bug"
    TASK = "task"
    EPIC = "epic"


class TaskStatus(str, Enum):
    BACKLOG = "backlog"
    TODO = "todo"
    IN_PROGRESS = "in_progress"
    IN_REVIEW = "in_review"
    DONE = "done"


class TaskPriority(str, Enum):
    LOW = "low"
    MEDIUM = "medium"
    HIGH = "high"
    CRITICAL = "critical"


class JiraSyncStatus(str, Enum):
    PENDING = "pending"
    SYNCED = "synced"
    FAILED = "failed"
    LOCAL_ONLY = "local_only"


FIBONACCI_POINTS: tuple[int, ...] = (1, 2, 3, 5, 8, 13, 21)


@dataclass
class Task:
    id: uuid.UUID
    project_id: uuid.UUID
    title: str
    description: str
    type: TaskType
    status: TaskStatus
    priority: TaskPriority
    estimated_points: Optional[int] = None
    actual_points: Optional[int] = None
    sprint_id: Optional[uuid.UUID] = None
    assignee_id: Optional[uuid.UUID] = None
    jira_key: Optional[str] = None
    jira_sync_status: JiraSyncStatus = JiraSyncStatus.LOCAL_ONLY
    tags: list[str] = field(default_factory=list)
    created_at: datetime = field(default_factory=datetime.utcnow)
    updated_at: datetime = field(default_factory=datetime.utcnow)

    def is_valid_story_points(self, points: int) -> bool:
        return points in FIBONACCI_POINTS

    def mark_done(self, actual_points: int) -> None:
        self.status = TaskStatus.DONE
        self.actual_points = actual_points
        self.updated_at = datetime.utcnow()
